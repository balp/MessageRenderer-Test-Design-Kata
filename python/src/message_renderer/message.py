import dataclasses


@dataclasses.dataclass(frozen=True)
class Message:
    header: str
    body: str
    footer: str
