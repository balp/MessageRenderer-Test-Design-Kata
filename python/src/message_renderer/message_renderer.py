from message_renderer.message import Message


class HeaderRenderer:
    def render(self, message: Message) -> str:
        return f"<h1>{message.header}</h1>"


class BodyRenderer:
    def render(self, message: Message) -> str:
        return f"<b>{message.body}</b>"


class FooterRenderer:
    def render(self, message: Message) -> str:
        return f"<i>{message.footer}</i>"


class MessageRenderer:
    def __init__(self):
        self._sub_renderers = [
            HeaderRenderer(),
            BodyRenderer(),
            FooterRenderer(),
        ]

    def sub_rendereres(self):
        return self._sub_renderers

    def render(self, message: Message) -> str:
        return ''.join(map(lambda x: x.render(message), self._sub_renderers))
