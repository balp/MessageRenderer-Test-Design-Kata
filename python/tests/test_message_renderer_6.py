from message_renderer.message import Message
from message_renderer.message_renderer import MessageRenderer, HeaderRenderer, BodyRenderer, FooterRenderer

from approvaltests.approvals import verify


def test_message_renderer_6():
    message = Message("header", "body", "footer")
    sut = MessageRenderer()
    html = sut.render(message)
    verify(html)
