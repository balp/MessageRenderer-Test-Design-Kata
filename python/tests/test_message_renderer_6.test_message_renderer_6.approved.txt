<h1>header</h1><b>body</b><i>footer</i>
